CONTRIBUTING.md

```diff
@@ -0,0 +1,82 @@
+# Contributing to Caliptra
+
+## Contributor License Agreement
+
+Per the [Charter](Caliptra_WG_Technical_Charter.md), contributors to Caliptra
+must have signed the CHIPS CLA.
+
+## Guidelines for code contributions
+
+* Changes must be made via pull request
+* [Branch protection rules](github_rules.md) ensure pull requests are reviewed
+  by project members
+* [Maintainers](MAINTAINERS.md) are expected to enforce `CODEOWNERS` rights
+  where necessary. Maintainers are also the sole owners of the repo's
+  configuration.
+* The branch protection rules include constraints to maintain a linear history
+  and avoid merges in pull requests.
+* Please keep a clean commit history, and no long series
+  of "fixup" patches. Use GitHub's "Squash and merge" option if necessary.
```

Comment:
Squash and Merge should be the default method for completing all Pull Requests back to main, but I would assume we don't actually care about commit history on the forked repos. As long as the Pull Request is squashed, the commit history at "main" will be clean.
Suggested rewording:
`In order to maintain a clean commit history on the "main" branch, all Pull Requests must be merged using the Squash strategy. Effort should be made to ensure Pull Requests are clean and pass required tests, to prevent a long series of subsequent Pull Requests with simple 'fixup' patches.`

Reply:
Applied!

Reply:
well I'm going to reopen this because I don't think "must squash and merge" is something to mandate

Reply:
If the PR merge itself is a squash, and reviewers trace a bug back to that PR, we should still be able to bisect further into the PR's individual commits without requiring each of those commits to be reflected onto the main repository, right? So this policy would just keep the main branch linear for readability, without really making debug an issue.

Reply:
(I also see your point with the more lenient reworded language, though it does look like the same sentence appears twice so I'd suggest cleaning that up)

Reply:
Oh thanks. I fat fingered in vim. Fixed (force pushed)